packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -212,6 +212,31 @@ class IOSDevice extends Device {
 
   DevicePortForwarder _portForwarder;
 
+  /// Uses `nslookup` to get the IP address of this device.
+  Future<String> getIP(bool ipv6) async {
+    // clean device name
+    final String cleanName = name
+        .replaceAll("'", '')
+        .replaceAll('’', '') // fancy apostrophe
+        .replaceAll(',', '')
+        .replaceAll(' ', '-');
+    final RegExp re = RegExp(r'^Address:\s+(.*)$', multiLine: true);
+    final RegExp error_re = RegExp(r'Non-authoritative answer:$', multiLine: true);
+    // Run nslookup
+    final String out = (await ProcessUtils.instance.run(
+      <String>[
+        'nslookup',
+        cleanName,
+      ],
+    )).stdout;
+    if (error_re.hasMatch(out)) {
+      return null;
+    }
+    // 1st match is IPv6, last match is IPv4
+    final String ip = (ipv6 ? re.firstMatch(out) : re.allMatches(out).last).group(1).trim();
```

Comment:
I'm still concerned about the unreliability of parsing this, cleaning the name up, etc.  There's no tests for this parsing, and we don't have confidence the nslookup formatting won't change, or that we have a complete understanding of what else this output can look like.

I started working on https://github.com/flutter/flutter/issues/54781 to get the logs and observatory URL from the lldb logs (which will also give us crash-on-launch logs and anything else that was lost in the iOS 13 universal logging transition).  Then we should be able to delete the FallbackDiscovery, and the iOS mDNS lookup code paths.